src/Linters/HHClientLinter.hack

```diff
@@ -18,8 +18,13 @@ final class HHClientLinter implements Linter {
   use LinterTrait;
   use SuppressibleTrait;
 
-  const type TConfig =
-    shape(?'ignore_except' => vec<int>, ?'ignore' => vec<int>, ...);
```

Comment:
I have added `ignoreExcept` as an alternative spelling of `ignore_except`.
I wanted to match the case of `lintMarkerAllowList` in global scope.

The following looks wrong:
```JSON
{
  "ignore_except": [],
  "lintMarkerAllowList": []
}
```

Reply:
The internal coding convention in Meta Platforms prefers lowercase_with_underscores for shape keys, although existing HHAST code base includes both lowerCamelCases and lowercase_with_underscores for shape keys.

https://github.com/hhvm/hhast/blob/d0a746ad3aabdfb053e7505aaf09583a8e813284/src/__Private/Resolution/get_uses_directly_in_scope.hack#L88

https://github.com/hhvm/hhast/blob/d0a746ad3aabdfb053e7505aaf09583a8e813284/src/Linters/DontDiscardNewExpressionsLinter.hack#L15

Reply:
I usually prefer snake_case as well, but I am trying to stay consistent with the hhast-lint.json format, which is 95% camelCase and 5% English with spaces.

### camelCase

`builtinLinters`, `extraLinters`, `disabledLinters`, `disabledAutoFixes`, `disableAllAutoFixes`, `disableAllLinters`, `linterConfigs`, `namespaceAliases`, `exceptionSuffixes`.

### English with spaces

`line ending`

The latter is a mistake on my part, but this can be fixed, since the linter doesn't observe the value for `"line ending"` (yet). So I can safely migrate this over to `lineEnding` without breaking BC.

Reply:
Fair enough. Let's document `hhast-lint.json` and `TConfig` about the state of lowerCamelCases usage.